tests/integration_tests/bugs/test_lp1835584.py

```diff
@@ -0,0 +1,63 @@
+""" Integration test for LP #1835584
+
+Kernels prior to 4.15 uppercased the dmi product_uuid, Azure
+relies on system UUID as their "instance-id". Upgrades to kernels >= 4.15
+will represent the same product_uuid as lowercase. System UUID should be a
+case-insensitive comparison.
+
+The test will launch an Ubuntu PRO FIPS image on Azure which has a 4.15 kernel.
+It updates cloud-init on that instance and boots first into the 4.15 kernel.
+Then install the linux-azure kernel which will install a 5.4 based kernel
+which will boot by default because it's version is greater than 4.15.
+
+Across the reboot, assert that we didn't re-run config_ssh by virtue of
+seeing only one semaphore creation log entry of type:
+
+ Writing to /var/lib/cloud/instances/<UUID>/sem/config_ssh -
+
+https://bugs.launchpad.net/cloud-init/+bug/1897099
+"""
+import re
+
+import pytest
+
+from tests.integration_tests.clouds import IntegrationCloud
+
+
+UBUNTU_PRO_FIPS_IMG = 'Canonical:0001-com-ubuntu-pro-bionic-fips:pro-fips-18_04'
+
+def _check_iid_insensitive_across_kernel_upgrade(instance):
+    uuid = instance.read_from_file('/sys/class/dmi/id/product_uuid')
+    assert uuid.isupper(), "UUID does not appear to be upppercase {}".format(
+      uuid
+    )
+    kernel = instance.read_from_file('uname -r').strip()
+    assert kernel == '4.15.0-1002-azure-fips'
```

Comment:
This will start failing once the Pro image and its kernel are revved, right?

Reply:
I've pinned the serial of the Ubuntu PRO bionic FIPS image so this won't break in this iteration of the integration test as we are only upgrading from stale FIPS -> newer kernel on Bionic.